src/Workleap.OpenApi.MSBuild/SpecGeneratorManager.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Microsoft.Build.Framework;
+
+namespace Workleap.OpenApi.MSBuild;
+
+internal class SpecGeneratorManager
+{
+    private readonly ILoggerWrapper _loggerWrapper;
+
+    public SpecGeneratorManager(ILoggerWrapper loggerWrapper)
+    {
+        this._loggerWrapper = loggerWrapper;
+    }
+
+    /// <summary>
+    /// Will overwrite the sourced-control specification files with the generated ones.
+    /// </summary>
+    public async Task UpdateSpecificationFilesAsync(IEnumerable<string> sourcedControlOpenApiSpecFiles, IEnumerable<string> generatedOpenApiSpecFiles, CancellationToken cancellationToken)
+    {
+        this._loggerWrapper.LogMessage("\n ******** Specification Generator: Updating specification files. ******** \n", MessageImportance.High);
+        
+        foreach (var baseSpecFile in sourcedControlOpenApiSpecFiles)
+        {
+            cancellationToken.ThrowIfCancellationRequested();
+            
+            var fileName = Path.GetFileName(baseSpecFile);
+            var generatedSpecFilePath = generatedOpenApiSpecFiles.FirstOrDefault(x => x.Contains(fileName));
```

Comment:
Same comment about how we compare file names here + we do enumerate twice. If we know the length then let's use a IReadonlycollection instead of IEnumerable to get rid of the warning

Reply:
uppdadted!